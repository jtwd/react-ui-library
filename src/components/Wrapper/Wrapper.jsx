import React from 'react'
import { node, bool } from 'prop-types'
import styled from 'styled-components'
import { md } from '../../theme/spacers'
import { textMaxWidth, shortMaxWidth } from '../../theme/units'

const StyledDiv = styled.div`
  padding: ${md};
  
  ${props => props.text && `
    max-width: ${textMaxWidth};
  `}
  
  ${props => props.short && `
    max-width: ${shortMaxWidth};
  `}
  
  ${props => props.centered && `
    margin-left: auto;
    margin-right: auto;
  `}
`

function Wrapper(props) {
  const { text, centered, short } = props
  return (
    <StyledDiv text={text} centered={centered} short={short}>
      {props.children}
    </StyledDiv>
  )
}

Wrapper.propTypes = {
  /** Sized appropriately for text */
  text: bool,
  /** Short Sized  */
  short: bool,
  /** Centered wrapper */
  centered: bool,
  /** Contents of wrapper */
  children: node
}

Wrapper.defaultProps = {
  text: false,
  short: false,
  centered: false,
  children: null
}

export default Wrapper
