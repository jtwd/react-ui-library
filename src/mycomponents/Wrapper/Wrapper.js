import styled from 'styled-components'
import { md } from '../../theme/spacers'

const Wrapper = styled.div`
  padding: ${md};
`

export default Wrapper
