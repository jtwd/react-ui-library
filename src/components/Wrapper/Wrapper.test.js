import React from 'react'
import {shallow} from 'enzyme'

import Wrapper from './Wrapper'

describe('Wrapper component', () => {
  it('renders a correctly by default', () => {
    const component = shallow(<Wrapper>Default</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders as a type of form', () => {
    const component = shallow(<Wrapper type="form">Form</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders as a type of main', () => {
    const component = shallow(<Wrapper type="main">Main</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders as a type of header', () => {
    const component = shallow(<Wrapper type="header">Header</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders as a type of footer', () => {
    const component = shallow(<Wrapper type="footer">Footer</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders as a type of section', () => {
    const component = shallow(<Wrapper type="section">Section</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders as a type of article', () => {
    const component = shallow(<Wrapper type="article">Article</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders a text wrapper', () => {
    const component = shallow(<Wrapper text>Text</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders a short wrapper', () => {
    const component = shallow(<Wrapper short>Text</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders a centered wrapper', () => {
    const component = shallow(<Wrapper centered>Centered</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders a centered text wrapper', () => {
    const component = shallow(<Wrapper text centered>Centered</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders a collapsed wrapper', () => {
    const component = shallow(<Wrapper collapse>Collapsed</Wrapper>)
    expect(component).toMatchSnapshot()
  })

  it('renders an app wrapper', () => {
    const component = shallow(<Wrapper app>App</Wrapper>)
    expect(component).toMatchSnapshot()
  })
})
