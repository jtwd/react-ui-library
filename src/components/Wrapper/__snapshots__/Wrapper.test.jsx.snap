// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper component renders a centered text wrapper 1`] = `
<styled.div
  app={false}
  centered={true}
  collapse={false}
  flex={null}
  middle={false}
  short={false}
  text={true}
>
  Centered
</styled.div>
`;

exports[`Wrapper component renders a centered wrapper 1`] = `
<styled.div
  app={false}
  centered={true}
  collapse={false}
  flex={null}
  middle={false}
  short={false}
  text={false}
>
  Centered
</styled.div>
`;

exports[`Wrapper component renders a collapsed wrapper 1`] = `
<styled.div
  app={false}
  centered={false}
  collapse={true}
  flex={null}
  middle={false}
  short={false}
  text={false}
>
  Collapsed
</styled.div>
`;

exports[`Wrapper component renders a correctly by default 1`] = `
<styled.div
  app={false}
  centered={false}
  collapse={false}
  flex={null}
  middle={false}
  short={false}
  text={false}
>
  Default
</styled.div>
`;

exports[`Wrapper component renders a short wrapper 1`] = `
<styled.div
  app={false}
  centered={false}
  collapse={false}
  flex={null}
  middle={false}
  short={true}
  text={false}
>
  Text
</styled.div>
`;

exports[`Wrapper component renders a text wrapper 1`] = `
<styled.div
  app={false}
  centered={false}
  collapse={false}
  flex={null}
  middle={false}
  short={false}
  text={true}
>
  Text
</styled.div>
`;

exports[`Wrapper component renders an app wrapper 1`] = `
<styled.div
  app={true}
  centered={false}
  collapse={false}
  flex={null}
  middle={false}
  short={false}
  text={false}
>
  App
</styled.div>
`;
